From 971b06d6004e6b65f853135e97e4f20b4bd16ba6 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 26 Mar 2024 14:34:27 +0100
Subject: [PATCH 0713/1695] plat-stm32mp2: sysconfig: simplify amcr check size

Simplify amcr check size.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ef87f09210a78898c71b372f6c29cb185a26206d plat-stm32mp2: sysconfig: simplify amcr check size]
Change-Id: Ifc0adcd18755bdf5d775c2021c00e550d1c565b1
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/369545
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../plat-stm32mp2/drivers/stm32mp25_syscfg.c   | 18 +++---------------
 1 file changed, 3 insertions(+), 15 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_syscfg.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_syscfg.c
index e24d389a8..70fc1bb37 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_syscfg.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_syscfg.c
@@ -146,35 +146,23 @@ void stm32mp25_syscfg_set_amcr(size_t mm1_size, size_t mm2_size)
 			    SYSCON_OFFSET(SYSCFG_OCTOSPIAMCR);
 	uint32_t amcr = 0;
 
+	if (mm1_size + mm2_size != SZ_256M)
+		panic();
+
 	switch (mm1_size) {
 	case 0:
-		if (mm2_size != SZ_256M)
-			panic();
-
 		amcr = SYSCFG_OCTOSPIAMCR_MM1_0_MM2_256;
 		break;
 	case SZ_64M:
-		if (mm2_size != SZ_192M)
-			panic();
-
 		amcr = SYSCFG_OCTOSPIAMCR_MM1_64_MM2_192;
 		break;
 	case SZ_128M:
-		if (mm2_size != SZ_128M)
-			panic();
-
 		amcr = SYSCFG_OCTOSPIAMCR_MM1_128_MM2_128;
 		break;
 	case SZ_192M:
-		if (mm2_size != SZ_64M)
-			panic();
-
 		amcr = SYSCFG_OCTOSPIAMCR_MM1_192_MM2_64;
 		break;
 	case SZ_256M:
-		if (mm2_size != 0)
-			panic();
-
 		amcr = SYSCFG_OCTOSPIAMCR_MM1_256_MM2_0;
 		break;
 	default:
-- 
2.47.2

